docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,313 @@
+
+Command requests in HTTP may also contain ds-pushevent and ds-returnevent query parameters (for GET commands).  These will be optionally provided key/value pairs represented in the message `envelope`'s query parameters (and optionally allows for other parameters in the future).
+
+``` JSON
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "API":"V2",
+    "queryParms": {
+        "ds-pushevent":"yes",
+        "ds-returnevent":"yes",
+     }
+     ...
+}
+```
+
+#### Message Payload
+
+The **request** message `payload` to the command service and those relayed to the device service would mimic their HTTP/REST request body alternatives.  The `payload` provides details needed in executing the command at the south side.
+
+In the example GET and PUT messages below, note the `envelope` wraps or encases the message `payload`.  The payload may be empty (as is typical of GET requests).
+
+``` JSON
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "API":"v2",
+    "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+    "queryParms": {
+        "ds-pushevent":"yes",
+        "ds-returnevent":"yes",
+     }
+    "payload": 
+    {
+    }
+}
+
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "API":"v2",
+    "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+    "payload": 
+    {
+        "AHU-TargetTemperature": "28.5",
+        "AHU-TargetBand": "4.0",
+        "AHU-TargetHumidity": {
+            "Accuracy": "0.2-0.3% RH",
+            "Value": 59
+        }
+    }
+}
+```
+
+
+The **response** message `payload` would contain the response from the south side, which is typically EdgeX event/reading objects (in the case of GET requests) but would also include any status code, error or service response details.
+
+Example response messages for a GET and PUT request are shown below.  Again, note that the message `envelope` wraps the response `payload`.
+
+``` JSON
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "apiVersion": "v2",
+    "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+    "statusCode": 0,
+    "payload": 
+    {
+    "message": "string",
+    "event": {
+        "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
+        "deviceName": "string",
+        "profileName": "string",
+        "created": 0,
+        "origin": 0,
+        "readings": [
+            "string"
+            ],
+        "tags": {
+            "Gateway-id": "HoustonStore-000123",
+            "Latitude": "29.630771",
+            "Longitude": "-95.377603"
+            }
+        }
+    }
+}
+
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "apiVersion": "v2",
+    "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+    "statusCode": 0,
+    "payload": 
+    {
+        "message": "string"
+    }
+}
+
+```
+
+!!! Note
+        Get command responses may include CBOR data.  The CBOR binary object would be enclosed in the payload.
```

Comment:
I think the point is that the payload is CBOR-encoded if it contains one or more binary readings from a device service. Otherwise it's JSON.

Reply:
Per Core WG discussion - others felt this was clear as stated.

Reply:
per the monthly architect's meeting of 3/28/22 - add something to the effect of:  The message envelop (has a content type indicator) will indicate that the payload is either CBOR or JSON.  The same message envelop content type indicator will be used in this message bus communications.